pynecone/middleware/hydrate_middleware.py

```diff
@@ -37,20 +39,42 @@ def preprocess(self, app: App, state: State, event: Event) -> Optional[Delta]:
                 load_event = None
 
             if load_event:
-                if isinstance(load_event, list):
-                    for single_event in load_event:
-                        self.execute_load_event(state, single_event)
-                else:
-                    self.execute_load_event(state, load_event)
+                if not isinstance(load_event, List):
+                    load_event = [load_event]
+                updates = []
+                for single_event in load_event:
+                    updates.append(
+                        await self.execute_load_event(
+                            state, single_event, event.token, event.payload
+                        )
+                    )
+                return updates
             return format.format_state({state.get_name(): state.dict()})
 
-    def execute_load_event(self, state: State, load_event: EventHandler) -> None:
+    async def execute_load_event(
+        self, state: State, load_event: EventHandler, token: str, payload: Dict
+    ) -> StateUpdate:
         """Execute single load event.
 
         Args:
             state: The client state.
             load_event: A single load event to execute.
+            token: client token
```

Comment:
nit: can we format the comments with uppercase and periods like "A client token." and "The event payload."